pandas/core/dtypes/cast.py

```diff
@@ -404,6 +411,25 @@ def maybe_promote(dtype, fill_value=np.nan):
     return dtype, fill_value
 
 
+def _check_lossless_cast(value, dtype: np.dtype) -> bool:
+    """
+    Check if we can cast the given value to the given dtype _losslesly_.
+
+    Parameters
+    ----------
+    value : object
+    dtype : np.dtype
+
+    Returns
+    -------
+    bool
+    """
+    casted = dtype.type(value)
+    if casted == value:
+        return True
+    return False
```

Comment:
Why not just `return dtype.type(value) == value` ?

Reply:
Either way would work.  I think in future steps this function is going to have to check a couple more things.